crates/wysiwyg/src/dom/nodes/dom_node.rs

```diff
@@ -311,6 +311,49 @@ where
             DomNode::Zwsp(_) => panic!("Can't slice a zwsp"),
         }
     }
+
+    /// Find the node based on its handle.
+    /// Panics if the handle is unset or invalid
+    pub fn lookup_node(&self, node_handle: &DomHandle) -> &DomNode<S> {
```

Comment:
I don't know if it would actually be better to have `split_sub_tree` return a new `Dom` object or have `Dom.lookup_node` actually just call `Dom.container().lookup_node()`. What do you think?

Reply:
I thought about this, but figured it'd be a bit confusing to have `Dom` contain sub-trees since they do not represent the whole document. But I may not be understanding the architecture here. Maybe @andybalaam can comment?

Reply:
Our `Dom` is just a structure that contains a tree and has helper methods for accessing it, so I think it would make sense to use one to temporarily host and modify the sub-tree. This would allow us to use helper `insert(&DomHandle)`, `remove(&DomHandle)`, etc., as well as look up, splitting, iterators, etc. 

As long as we make clear that the returned Dom is a sub-tree and not the original one I'd say it's probably fine to use a Dom object.

Reply:
Ok, that makes sense. I just tried to refactor this to replace `DomNode` with `Dom` but I was then unable to make it work without undoing the improvement in [`40be2d1`](https://github.com/matrix-org/matrix-rich-text-editor/pull/415/commits/40be2d1e2ece87068303169572873d96312f6ed2) (i.e. I needed to clone the subtree `Dom` children in order to move them into the new parent node).

If you know a way to do this work please let me know and I'll make a follow-up PR with this

Reply:
Oh, sorry. I didn't mean that you should do it now, I was just arguing that we should probably make that change in the future. Since I already modified `split_sub_tree` on my other PR I'll try to do this change myself, then adapt existing code to it.